### PR TITLE
XW-4097 refactor mobile-wiki and mercury tests

### PR DESCRIPTION
**Pull request by qaga**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 14 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CategoryPageTest.java#L107](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L107): This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L151](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L151): Move the "prod" string literal on the left side of this string comparison. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [HeaderTest.java#L104](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L104): Move the contents of this initializer to a standard constructor or to field initializers. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1171)
1. ![MAJOR][MAJOR] [HeaderTest.java#L108](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L108): Move this closing curly brace to the next line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
1. ![MINOR][MINOR] [CategoryPageTest.java#L25](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L25): Define a constant instead of duplicating this literal "Category:%s" 6 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L28](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L28): Define a constant instead of duplicating this literal "[[%s]]" 7 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L44](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L44): Define a constant instead of duplicating this literal "some irrelevant content of category article" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CommentsTests.java#L76](https://github.com/Wikia/selenium-tests/blob/b42f0a2b1fe11ed2772c3e7cd1903935c8420556/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CommentsTests.java#L76): Define a constant instead of duplicating this literal "is correct" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 14 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CategoryPageTest.java#L107](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L107): This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L151](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L151): Move the "prod" string literal on the left side of this string comparison. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [HeaderTest.java#L104](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L104): Move the contents of this initializer to a standard constructor or to field initializers. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1171)
1. ![MAJOR][MAJOR] [HeaderTest.java#L108](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L108): Move this closing curly brace to the next line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
1. ![MINOR][MINOR] [CategoryPageTest.java#L25](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L25): Define a constant instead of duplicating this literal "Category:%s" 6 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L28](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L28): Define a constant instead of duplicating this literal "[[%s]]" 7 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L44](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L44): Define a constant instead of duplicating this literal "some irrelevant content of category article" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CommentsTests.java#L76](https://github.com/Wikia/selenium-tests/blob/7177bfa1462dc5e461e46aab9b15211558a79bc4/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CommentsTests.java#L76): Define a constant instead of duplicating this literal "is correct" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationMercuryTests.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
 import com.wikia.webdriver.common.skin.Skin;
 import com.wikia.webdriver.elements.mercury.pages.ArticlePage;
 import com.wikia.webdriver.elements.mercury.pages.discussions.GuidelinesPage;
+import com.wikia.webdriver.testcases.mobilewikitests.NavigationTests;
```
Comment (rybmat) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/NavigationMercuryTests.java]: is it necessary?

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchMercuryTests.java [hunk 2]
```diff
@@ -12,6 +12,8 @@
 import com.wikia.webdriver.elements.mercury.pages.SearchResultsPage;
 import com.wikia.webdriver.elements.mercury.pages.discussions.GuidelinesPage;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject;
+import com.wikia.webdriver.testcases.mobilewikitests.SearchTests;
```
Comment (rybmat) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchMercuryTests.java]: is it necessary?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 14 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CategoryPageTest.java#L107](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L107): This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L151](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L151): Move the "prod" string literal on the left side of this string comparison. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [HeaderTest.java#L104](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L104): Move the contents of this initializer to a standard constructor or to field initializers. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1171)
1. ![MAJOR][MAJOR] [HeaderTest.java#L108](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L108): Move this closing curly brace to the next line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
1. ![MINOR][MINOR] [CategoryPageTest.java#L25](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L25): Define a constant instead of duplicating this literal "Category:%s" 6 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L28](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L28): Define a constant instead of duplicating this literal "[[%s]]" 7 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L44](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L44): Define a constant instead of duplicating this literal "some irrelevant content of category article" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CommentsTests.java#L76](https://github.com/Wikia/selenium-tests/blob/e90121728152d7ebbf8cda37621e052c14866290/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CommentsTests.java#L76): Define a constant instead of duplicating this literal "is correct" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 14 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [CategoryPageTest.java#L107](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L107): This block of commented-out lines of code should be removed. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L151](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L151): Move the "prod" string literal on the left side of this string comparison. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1132)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HTMLTitleTests.java#L164](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HTMLTitleTests.java#L164): Either remove or fill this block of code. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [HeaderTest.java#L104](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L104): Move the contents of this initializer to a standard constructor or to field initializers. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1171)
1. ![MAJOR][MAJOR] [HeaderTest.java#L108](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/HeaderTest.java#L108): Move this closing curly brace to the next line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
1. ![MINOR][MINOR] [CategoryPageTest.java#L25](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L25): Define a constant instead of duplicating this literal "Category:%s" 6 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L28](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L28): Define a constant instead of duplicating this literal "[[%s]]" 7 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CategoryPageTest.java#L44](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CategoryPageTest.java#L44): Define a constant instead of duplicating this literal "some irrelevant content of category article" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [CommentsTests.java#L76](https://github.com/Wikia/selenium-tests/blob/ec788193b3b59f9fbce58b62967e5c11a4f30c2e/src/test/java/com/wikia/webdriver/testcases/mobilewikitests/CommentsTests.java#L76): Define a constant instead of duplicating this literal "is correct" 3 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**